### DISCUSSION TITLE
Offline reading

### DISCUSSION THREAD

**Discussion by andreyorst** [jocmp/capyreader#550] [Feature Requests]
I'd like to requirest a feature: offline reading

For a long time I was using the Feeder app, and I liked it a lot because it had almost all of the features I needed. It has some problems, most of which are non-existent in Capy Reader. However, I do miss some features.

In Feeder, there's an option to estimate time it takes to read an article. While I don't think it's practically useful metric, it has a side effect of fetching the article content while syncing, thus articles are cached in the app and are available for offline reading.

I'd like to ask for this feature, because I have a limited amount of traffic on my phone, and I often download articles while I'm at home, and read them in the subway while going to work.  With Capy it's not quite possible at the moment, unless I manually open each article beforehand.

Do you think this feature could be a part of Capy Reader?

**Comment by jocmp** [comment]
Hey @andreyorst - Capy supports offline articles but not full offline content. Just to clarify, is your suggestion to download full content offline?

**Reply by andreyorst** [reply]
Yes. Most of articles in my feed require "full content" switch, so perhaps I didn't notice that the feature is partially there

**Reply by jocmp** [reply]
One more question for you. Would this feature automatically download new articles or just ones you've asked for it to save?

**Reply by andreyorst** [reply]
So here's how it works in Feeder:

https://github.com/user-attachments/assets/f1e8efd9-1a20-4bb2-806b-20ba0f316c8e

You can see that the update process stops (the spinner disappears) and then each article updates by showing its estimated read time. This has a side effect of downloading full content, and that's basically what I want.

So in short: all articles

**Reply by alanvww** [reply]
Hi @jocmp! How is the implementation of this feature going? I would also suggest considering this feature as an "offline cache" for loaded articles. And perhaps 3-5 latest articles per source in one complete refresh would be helpful to unstable internet conditions.

**Reply by jocmp** [reply]
Hey @alanvww, I haven't worked on this feature yet. Though the more I've thought about it, the less I'm sure there's a one-size-fits-all solution. At this point in time, I think read-it-later services like [Readeck](https://readeck.org/) are a much better option for offline articles still.

**Reply by alanvww** [reply]
> Hey @alanvww, I haven't worked on this feature yet. Though the more I've thought about it, the less I'm sure there's a one-size-fits-all solution. At this point in time, I think read-it-later services like [Readeck](https://readeck.org/) are a much better option for offline articles still.

Thanks for your input on this.

Building on that, my main thought was to first address the issue where the app crashes if I try to load an article without internet. (It might be a bug on my device :/ ? )

My idea for 'offline reading' is more of a simple fallback. We could cache a few articles on the device for a longer duration. That way, the app wouldn't have to reload them every time, and more importantly, it would have something to show the user if they're offline, instead of crashing. This wouldn't be full offline support, but rather a way to handle that specific 'no internet' state gracefully.

**Reply by jocmp** [reply]
Just to quick check on the bug though, which version are you on? I think I found a bug in 2025.06.1043 that crashes on offline reading. Hopefully that should be fixed in the latest I pushed today.

**Reply by alanvww** [reply]
> Just to quick check on the bug though, which version are you on? I think I found a bug in 2025.06.1043 that crashes on offline reading. Hopefully that should be fixed in the latest I pushed today. 

I am using Capy Reader 2025.06.1143. I cleared the storage seems able to fix that, but have another issue that when extracting full content on a certain source, the app will crash. In my case, it's NYTimes.

**Reply by Wurftaube** [reply]
I just switched from **spaRSS** (https://github.com/Etuldan/spaRSS unfortunately, the repository has been archived since 24. Dec. 2022) to **Capy Reader** 💟 after neither Feeder nor any other RSS reader appealed to me.
SpaRSS offers excellent options for editing and sorting feeds, including the option to "_retrieve full text_" for each feed. 
This has worked perfectly for me for years.
![Screenshot_20251209_183758_spaRSS](https://github.com/user-attachments/assets/d81926fb-9c72-474c-9e71-95be3773f32c)

**Reply by jocmp** [reply]
Thanks for the links, I'll take a look!

**Comment by jhench** [comment]
Bump to this request. Was recently traveling with minimal access and was hoping I could star a post to get it to download a cache or fetch full article (like pocket or I'm assuming OPs request).
